src/Dockerfile

```diff
@@ -96,10 +95,22 @@ ENV BASH_ENV=${CONTAINER_SCRIPTS_PATH}/scl_enable \
 {% endif %}
 VOLUME ["/var/lib/pgsql/data"]
 
-# {APP_DATA} needs to be accessed by postgres user while s2i assembling
-# postgres user changes permissions of files in APP_DATA during assembling
-RUN /usr/libexec/fix-permissions ${APP_DATA} && \
-    usermod -a -G root postgres
+# S2I permission fixes
+# --------------------
+# 1. unless specified otherwise (or - equivalently - we are in OpenShift), s2i
+#    build process would be executed as 'uid=26(postgres) gid=26(postgres)'.
+#    Such process wouldn't be able to execute the default 'assemble' script
+#    correctly (it transitively executes 'fix-permissions' script).  So let's
+#    add the 'postgres' user into 'root' group here
+#
+# 2. we call fix-permissions on $APP_DATA here directly (UID=0 during build
+#    anyways) to assure that s2i process is actually able to _read_ the
+#    user-specified scripting.
+#
+# WARNING: this doesn't work in rootless s2i build because editing /etc/group
+# has no effect there (processes are never able to call set*id syscalls).
```

Comment:
This WARNING probably isn't valid.  I'm able to `su - ..` now in rootless container.